docroot/modules/custom/mass_caching/mass_caching.module

```diff
@@ -130,3 +155,17 @@ function mass_caching_entity_insert(EntityInterface $entity) {
     $purger->purgePath($path);
   }
 }
+
+/**
+ * Determine whether a file uri is private.
+ *
+ * @param string $uri
+ *   A stream, referenced as "scheme://target" or "data:target".
+ *
+ * @return bool
+ *   TRUE if the uri is private FALSE if it is anything else.
+ */
+function _mass_caching_uri_is_private($uri) {
+  $file_system = \Drupal::service('file_system');
+  return strpos($file_system->uriScheme($uri), 'public');
```

Comment:
Isn't this the opposite condition from what we want (returns true if it's public)?  Also, let's do this the official way:
```
$stream_wrapper_manager = \Drupal::service('stream_wrapper_manager');
$destination_scheme = $stream_wrapper_manager::getScheme($file->getFileUri());
return $destination_scheme === 'private';
```

Reply:
Done